format/formatter_test.go

```diff
@@ -47,3 +52,227 @@ func TestInvalidFormatterName(t *testing.T) {
 		as.ErrorIs(err, format.ErrInvalidName)
 	}
 }
+
+func TestFormatterHash(t *testing.T) {
```

Comment:
I feel like this test would get a lot shorter if we introduced a helper method. Something like:

```go
func assertHashChangeAndStable(f *formatter, oldHash int) (newHash int) {
	newHash, err := f.Hash()
	as.NoError(err)

	as.NotEqual(oldHash, newHash, "hash should have changed")

	// hash again without making changes
	newHash2, err := f.Hash()
	as.NoError(err)
	as.Equal(newHash, newHash2, "hash should not have changed")

	return newHash
}
```

Reply:
Yeah, fair enough. Was being lazy :)